tests/executors/test_local_executor.py

```diff
@@ -49,11 +48,7 @@ def execution_parallelism(self, parallelism=0):
 
         executor.running['fail'] = True
```

Comment:
I think this also should be moved before execute_async above. I also do not understand why we need the try/except